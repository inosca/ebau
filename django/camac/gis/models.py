from django.db import models
from django.utils.translation import gettext_lazy as _


class GISConfig(models.Model):
    CLIENT_SOGIS = "camac.gis.clients.sogis.SoGisClient"
    CLIENT_PARAM = "camac.gis.clients.param.ParamClient"

    CLIENT_CHOICES = [
        (CLIENT_SOGIS, _("GIS Canton Solothurn")),
        (CLIENT_PARAM, _("Parameter")),
    ]

    slug = models.SlugField(primary_key=True)
    client = models.CharField(max_length=255, choices=CLIENT_CHOICES)
    config = models.JSONField(default=dict)
    disabled = models.BooleanField(default=False)
