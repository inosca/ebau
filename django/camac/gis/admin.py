from adminsortable2.admin import SortableAdminMixin
from django.contrib.admin import ModelAdmin, register
from django.db.models import JSONField
from django_json_widget.widgets import JSONEditorWidget

from camac.admin import EbauAdminMixin
from camac.gis.models import GISDataSource


@register(GISDataSource)
class GISDataSourceAdmin(EbauAdminMixin, SortableAdminMixin, ModelAdmin):
    list_display = ["description", "client", "disabled"]
    list_per_page = 50
    search_fields = ["description", "client"]
    formfield_overrides = {
        JSONField: {"widget": JSONEditorWidget},
    }
