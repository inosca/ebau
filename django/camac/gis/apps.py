from django.apps import AppConfig


class GisConfig(AppConfig):
    default_auto_field = "django.db.models.BigAutoField"
    name = "camac.gis"
