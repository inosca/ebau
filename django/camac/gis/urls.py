from django.urls import re_path

from camac.gis import views

urlpatterns = [
    re_path(
        r"^data/(?:(?P<task_id>[\w-]+)/)?$",
        views.GISDataView.as_view(),
        name="gis-data",
    ),
    re_path(
        r"^apply$",
        views.GISApplyView.as_view(),
        name="gis-apply",
    ),
]
