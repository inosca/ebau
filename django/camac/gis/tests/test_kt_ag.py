import json

import pytest
from caluma.caluma_form.models import Question
from django.conf import settings
from django.core.management import call_command
from django.urls import reverse
from rest_framework import status

from camac.gis.models import GISDataSource

TEST_SCENARIOS = [
    {
        # Grundinformationen (Adresse, Grundstück, Zone, etc.)
        "coords": (2635082.564072413, 1241236.536643672),
        "checked_questions": [
            "gemeinde",
            "ort-grundstueck",
            "parzelle",
        ],
    },
    {
        # Waldnaturschutzinventar
        "coords": (2657749, 1240978),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Landschaftsschutzdekrete (detailliert Legende)
        "coords": (2657686.7954999995, 1241576.6492766952),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # BLN - Bundesinventar der Landschaften und Naturdenkmäler von nationaler Bedeutung
        "coords": (2657645.3687233045, 1241677.0886165234),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Wildtierkorridore: Perimeter und Warteräume
        "coords": (2657308, 1241300),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Wasser-und Zugvogelreservate
        "coords": (2670201, 1242283),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Trockenwiesen
        "coords": (2645903, 1252830),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Amphibienlaichgebiet - ortsfeste Objekte
        "coords": (2647143, 1250753),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Amphibienlaichgebiet - Wanderobjekte
        "coords": (2650151, 1244549),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Hecke
        "coords": (2650119, 1244361),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Kantonale Denkmalschutzobjekte
        "coords": (2654406.245488748, 1243184.0781744092),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Geschützte Naturobjekte im Kulturlandplan
        "coords": (2650229, 1238861),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Geschützte Kulturobjekte im Kulturlandplan
        "coords": (2650103, 1238632),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Naturschutzgebiet von kant. Bedeutung (NkB) Richtplan L 2.5
        "coords": (2663425, 1247037),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Bundesinventar der Flachmoore von nationaler Bedeutung
        "coords": (2663415, 1247092),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Weilerzone
        "coords": (2652240, 1243774),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Auengebiete von nationaler Bedeutung
        "coords": (2670175, 1242187),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Bundesinventar der Hoch- und Übergangsmoore von nationaler Bedeutung
        "coords": (2665946, 1245959),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Überlagerte Schutzräume für Lebensräume und Landschaften (ohne Gewässerraum)
        "coords": (2654826, 1243271),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Bauinventarobjekte
        "coords": (2654357, 1241392),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Gebäude unter Schutz
        "coords": (2654391, 1241377),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Kurzinventar Denkmalpflege
        "coords": (2655777, 1242133),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Archäologische Fundstellen nach Typ, mit Denkmalschutz
        "coords": (2654373, 1242470),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Kantonale Denkmalschutzobjekte
        "coords": (2654406, 1243184),
        "checked_questions": ["ueberlagerte-schutzzonen-und-schutzobjekte"],
    },
    {
        # Hochwasser, Gefahrenzone Hochwasser, gewaesserschutzbereich
        "coords": (2654643, 1244046),
        "checked_questions": [
            "plan-der-gefahrenkommission",
            "das-bauvorhaben-befindet-sich-in",
        ],
    },
    {
        # Bauzonen: Grundnutzung
        "coords": (2654743, 1243772),
        "checked_questions": ["zonenplan"],
    },
    {
        # Permanenter Rutsch, gefahrenzone-massenbewegungen
        "coords": (2642903, 1240177),
        "checked_questions": [
            "plan-der-gefahrenkommission",
            "das-bauvorhaben-befindet-sich-in",
        ],
    },
    {
        # Assekuranznummer
        "coords": (2643609, 1241663),
        "checked_questions": ["gebaeude", "amtliche-gebaeudenummer", "egid-nr"],
    },
    {
        # Kantonaler Nutzungsplan Verkehr
        "coords": (2635722, 1239843),
        "checked_questions": ["gestaltungs-und-erschliessungsplan"],
    },
    {
        # Gewässerschutzbereich Au, Eisenbahnlinie
        "coords": (2635666, 1240642),
        "checked_questions": ["das-bauvorhaben-befindet-sich-in"],
    },
    {
        # grundwasserschutzzone
        "coords": (2636545, 1239084),
        "checked_questions": ["das-bauvorhaben-befindet-sich-in"],
    },
    {
        # Bachkataster
        "coords": (2636602, 1239145),
        "checked_questions": ["das-bauvorhaben-befindet-sich-in"],
    },
    {
        # Gewässerraum
        "coords": (2636488, 1239256),
        "checked_questions": ["das-bauvorhaben-befindet-sich-in"],
    },
    {
        # Chemiesicherheit
        "coords": (2636719, 1239492),
        "checked_questions": ["das-bauvorhaben-befindet-sich-in"],
    },
    {
        # Risikokataster: Konsultationsbereich Gasleitung : Chemiesicherheit
        "coords": (2636488, 1239256),
        "checked_questions": ["das-bauvorhaben-befindet-sich-in"],
    },
    {
        # Kantons- oder Nationalstrassen
        "coords": (2636625, 1239426),
        "checked_questions": ["das-bauvorhaben-befindet-sich-in", "kantonsstrassen"],
    },
    {
        # Waldgrenze
        "coords": (2635317, 1238615),
        "checked_questions": ["waldareal"],
    },
    {
        # Prüfperimeter Bodenaushub
        "coords": (2634417, 1240019),
        "checked_questions": ["das-bauvorhaben-befindet-sich-in"],
    },
    {
        # Kataster belasteter Standorte
        "coords": (2634847, 1238635),
        "checked_questions": ["das-bauvorhaben-befindet-sich-in"],
    },
]


@pytest.fixture
def ag_data_sources(
    question_factory, question_option_factory, option_factory, mock_municipalities
):
    gis_questions = [
        ("gemeinde", Question.TYPE_DYNAMIC_CHOICE),
        ("ort-grundstueck", Question.TYPE_TEXT),
        ("parzellen", Question.TYPE_TABLE),
        ("parzellennummer", Question.TYPE_TEXT),
        ("e-grid-nr", Question.TYPE_TEXT),
        (
            "waldareal",
            Question.TYPE_CHOICE,
            ["ja", "waldabstandsbereich", "nein"],
        ),
        ("kantonsstrassen", Question.TYPE_CHOICE, ["ja", "nein"]),
        ("denkmalschutzobjekt", Question.TYPE_CHOICE, ["ja", "nein"]),
        ("ausserhalb-bauzone", Question.TYPE_CHOICE, ["ja", "nein"]),
        (
            "das-bauvorhaben-befindet-sich-in",
            Question.TYPE_MULTIPLE_CHOICE,
            [
                "gewaesserraum",
                "gewaesserschutzbereich",
                "grundwasserschutzzone",
                "kataster-belasteter-standorte",
                "pruefperimeter-bodenaushub",
                "eisenbahnlinie",
                "archaeologische-fundstelle",
                "gefahrenzone-massenbewegungen",
                "gefahrenzone",
                "risikokataster-chemiesicherheit",
            ],
        ),
    ]

    for config in gis_questions:
        slug = config[0]
        type = config[1]
        q = question_factory(slug=slug, type=type, label=slug)
        if len(config) == 3:
            for i, option in enumerate(reversed(config[2])):
                question_option_factory(
                    question=q,
                    option=option_factory(slug=f"{slug}-{option}", label=option),
                    sort=i,
                )

    Question.objects.filter(slug="gemeinde").update(data_source="Municipalities")
    mock_municipalities(["Aarburg", "Woanders"])

    return GISDataSource.objects.all()


@pytest.fixture
def ag_config(gis_data_source_factory, question_factory):
    call_command("loaddata", settings.ROOT_DIR("kt_ag/config/gis.json"))

    return GISDataSource.objects.all()


@pytest.mark.parametrize(
    "scenario",
    TEST_SCENARIOS,
)
@pytest.mark.vcr()
def test_ag_client(
    db,
    admin_client,
    gis_snapshot,
    vcr_config,
    ag_config,
    scenario,
    ag_data_sources,
):
    x, y = scenario["coords"]
    response = admin_client.get(
        reverse("gis-data"),
        data={
            "query": json.dumps(
                {
                    "markers": [{"x": x, "y": y}],
                    "geometry": "POINT",
                }
            ),
            "form": "baugesuch",
        },
    )

    assert response.status_code == status.HTTP_200_OK
    checked_data = {
        k: v
        for k, v in response.json()["data"].items()
        if k in scenario["checked_questions"]
    }

    assert checked_data == gis_snapshot
