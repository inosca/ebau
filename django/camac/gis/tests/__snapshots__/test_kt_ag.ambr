# serializer version: 1
# name: test_ag_client[scenario0]
  dict({
    'gemeinde': dict({
      'displayValue': 'Aarburg',
      'hidden': False,
      'label': 'gemeinde',
      'value': '1',
    }),
    'ort-grundstueck': dict({
      'hidden': False,
      'label': 'ort-grundstueck',
      'value': 'Aarburg',
    }),
    'parzelle': dict({
      'hidden': False,
      'label': None,
      'value': list([
        dict({
          'e-grid-nr': dict({
            'label': 'e-grid-nr',
            'value': 'CH607706874573',
          }),
          'gemeinde': dict({
            'displayValue': 'Aarburg',
            'label': 'gemeinde',
            'value': '1',
          }),
          'parzellennummer': dict({
            'label': 'parzellennummer',
            'value': '646',
          }),
        }),
      ]),
    }),
  })
# ---
# name: test_ag_client[scenario10]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Hochstammobstbestand, Kantonales Denkmalschutzobjekt (Retterswil 8, Meierhof)',
    }),
  })
# ---
# name: test_ag_client[scenario11]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Geschütztes Naturobjekt im Kulturlandplan (Einzelbaum, geschützt), Landschaftsschutzzone',
    }),
  })
# ---
# name: test_ag_client[scenario12]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Geschütztes Kulturobjekt im Kulturlandplan (Brunnen, geschützt), Landschaftsschutzzone',
    }),
  })
# ---
# name: test_ag_client[scenario13]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Amphibienlaichgebiet von nationaler Bedeutung, Naturschutzgebiet von kant. Bedeutung (NkB), Flachmoor von nationaler Bedeutung, Archäologische Fundstelle (52(B)12)',
    }),
  })
# ---
# name: test_ag_client[scenario14]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Amphibienlaichgebiet von nationaler Bedeutung, Naturschutzgebiet von kant. Bedeutung (NkB), Flachmoor von nationaler Bedeutung, Archäologische Fundstelle (52(B)12)',
    }),
  })
# ---
# name: test_ag_client[scenario15]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Hecke im Kulturlandplan, Weilerzone, Archäologische Fundstelle (6(B)10 b)',
    }),
  })
# ---
# name: test_ag_client[scenario16]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Übrige Zonen innerhalb Dekretsgebiet (kant. NP), Landschaften und Naturdenkmäler von nationaler Bedeutung (BLN), Waldnaturschutzinventar (Moos), Wasser- und Zugvogelreservate von internationaler und nationaler Bedeutung, Amphibienlaichgebiet von nationaler Bedeutung, Auengebiet von nationaler Bedeutung',
    }),
  })
# ---
# name: test_ag_client[scenario17]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Waldnaturschutzinventar (Fischbacher Moos), Wildtierkorridor (AG-R09), Amphibienlaichgebiet von nationaler Bedeutung, Naturschutzgebiet von kant. Bedeutung (NkB), Flachmoor von nationaler Bedeutung, Hoch- oder Übergangsmoor von nationaler Bedeutung',
    }),
  })
# ---
# name: test_ag_client[scenario18]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Wildtierkorridor (AG-R13), Vernetzungszone, Hochstammobstbestand, Landschaftsschutzzone',
    }),
  })
# ---
# name: test_ag_client[scenario19]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Bauinventarobjekt (Leutwilerstrasse 2), Gebäude mit Substanzschutz, Archäologische Fundstelle (100(A)7)',
    }),
  })
# ---
# name: test_ag_client[scenario1]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Naturschutzzone innerhalb Dekretsgebiet (kant. NP), Landschaften und Naturdenkmäler von nationaler Bedeutung (BLN), Waldnaturschutzinventar (Moordorf Risle), Amphibienlaichgebiet von nationaler Bedeutung, Naturschutzgebiet von kant. Bedeutung (NkB), Naturwaldgemässe Bestockung, Archäologische Fundstelle (154(A)1)',
    }),
  })
# ---
# name: test_ag_client[scenario20]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Bauinventarobjekt (Schulhaus), Gebäude mit Substanzschutz, Archäologische Fundstelle (100(A)7)',
    }),
  })
# ---
# name: test_ag_client[scenario21]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Kurzinventar Denkmalpflege (Alte Post)',
    }),
  })
# ---
# name: test_ag_client[scenario22]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Hecke im Kulturlandplan, Landschaftsschutzzone, Archäologische Fundstelle (100(A)3)',
    }),
  })
# ---
# name: test_ag_client[scenario23]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Hochstammobstbestand, Kantonales Denkmalschutzobjekt (Retterswil 8, Meierhof)',
    }),
  })
# ---
# name: test_ag_client[scenario24]
  dict({
    'das-bauvorhaben-befindet-sich-in': dict({
      'hidden': False,
      'label': 'das-bauvorhaben-befindet-sich-in',
      'value': list([
        dict({
          'displayValue': 'gewaesserraum',
          'value': 'das-bauvorhaben-befindet-sich-in-gewaesserraum',
        }),
        dict({
          'displayValue': 'gewaesserschutzbereich',
          'value': 'das-bauvorhaben-befindet-sich-in-gewaesserschutzbereich',
        }),
        dict({
          'displayValue': 'gefahrenzone',
          'value': 'das-bauvorhaben-befindet-sich-in-gefahrenzone',
        }),
      ]),
    }),
    'plan-der-gefahrenkommission': dict({
      'hidden': False,
      'label': None,
      'value': 'Hochwasser (Restgefährdung), Hochwasser (geringe Gefährdung), Hochwasser (mittlere Gefährdung)',
    }),
  })
# ---
# name: test_ag_client[scenario25]
  dict({
    'zonenplan': dict({
      'hidden': False,
      'label': None,
      'value': 'Wohnzone 2',
    }),
  })
# ---
# name: test_ag_client[scenario26]
  dict({
    'das-bauvorhaben-befindet-sich-in': dict({
      'hidden': False,
      'label': 'das-bauvorhaben-befindet-sich-in',
      'value': list([
        dict({
          'displayValue': 'gewaesserschutzbereich',
          'value': 'das-bauvorhaben-befindet-sich-in-gewaesserschutzbereich',
        }),
        dict({
          'displayValue': 'gefahrenzone-massenbewegungen',
          'value': 'das-bauvorhaben-befindet-sich-in-gefahrenzone-massenbewegungen',
        }),
      ]),
    }),
    'plan-der-gefahrenkommission': dict({
      'hidden': False,
      'label': None,
      'value': 'Permanente Rutschung (permanenter Rutsch)',
    }),
  })
# ---
# name: test_ag_client[scenario27]
  dict({
    'gebaeude': dict({
      'hidden': False,
      'label': None,
      'value': list([
        dict({
          'amtliche-gebaeudenummer': dict({
            'label': None,
            'value': '754',
          }),
          'egid-nr': dict({
            'label': None,
            'value': '613891',
          }),
        }),
      ]),
    }),
  })
# ---
# name: test_ag_client[scenario28]
  dict({
    'gestaltungs-und-erschliessungsplan': dict({
      'hidden': False,
      'label': None,
      'value': 'KNP K204 Wiggertalstrasse Aarburg Oftringen Zofingen',
    }),
  })
# ---
# name: test_ag_client[scenario29]
  dict({
    'das-bauvorhaben-befindet-sich-in': dict({
      'hidden': False,
      'label': 'das-bauvorhaben-befindet-sich-in',
      'value': list([
        dict({
          'displayValue': 'gewaesserschutzbereich',
          'value': 'das-bauvorhaben-befindet-sich-in-gewaesserschutzbereich',
        }),
        dict({
          'displayValue': 'pruefperimeter-bodenaushub',
          'value': 'das-bauvorhaben-befindet-sich-in-pruefperimeter-bodenaushub',
        }),
        dict({
          'displayValue': 'eisenbahnlinie',
          'value': 'das-bauvorhaben-befindet-sich-in-eisenbahnlinie',
        }),
        dict({
          'displayValue': 'risikokataster-chemiesicherheit',
          'value': 'das-bauvorhaben-befindet-sich-in-risikokataster-chemiesicherheit',
        }),
      ]),
    }),
  })
# ---
# name: test_ag_client[scenario2]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Landschaftsschutzzone innerhalb Dekretsgebiet (kant. NP), Landschaften und Naturdenkmäler von nationaler Bedeutung (BLN)',
    }),
  })
# ---
# name: test_ag_client[scenario30]
  dict({
    'das-bauvorhaben-befindet-sich-in': dict({
      'hidden': False,
      'label': 'das-bauvorhaben-befindet-sich-in',
      'value': list([
        dict({
          'displayValue': 'gewaesserschutzbereich',
          'value': 'das-bauvorhaben-befindet-sich-in-gewaesserschutzbereich',
        }),
        dict({
          'displayValue': 'grundwasserschutzzone',
          'value': 'das-bauvorhaben-befindet-sich-in-grundwasserschutzzone',
        }),
        dict({
          'displayValue': 'pruefperimeter-bodenaushub',
          'value': 'das-bauvorhaben-befindet-sich-in-pruefperimeter-bodenaushub',
        }),
        dict({
          'displayValue': 'risikokataster-chemiesicherheit',
          'value': 'das-bauvorhaben-befindet-sich-in-risikokataster-chemiesicherheit',
        }),
      ]),
    }),
  })
# ---
# name: test_ag_client[scenario31]
  dict({
    'das-bauvorhaben-befindet-sich-in': dict({
      'hidden': False,
      'label': 'das-bauvorhaben-befindet-sich-in',
      'value': list([
        dict({
          'displayValue': 'gewaesserraum',
          'value': 'das-bauvorhaben-befindet-sich-in-gewaesserraum',
        }),
        dict({
          'displayValue': 'gewaesserschutzbereich',
          'value': 'das-bauvorhaben-befindet-sich-in-gewaesserschutzbereich',
        }),
        dict({
          'displayValue': 'grundwasserschutzzone',
          'value': 'das-bauvorhaben-befindet-sich-in-grundwasserschutzzone',
        }),
        dict({
          'displayValue': 'pruefperimeter-bodenaushub',
          'value': 'das-bauvorhaben-befindet-sich-in-pruefperimeter-bodenaushub',
        }),
        dict({
          'displayValue': 'risikokataster-chemiesicherheit',
          'value': 'das-bauvorhaben-befindet-sich-in-risikokataster-chemiesicherheit',
        }),
      ]),
    }),
  })
# ---
# name: test_ag_client[scenario32]
  dict({
    'das-bauvorhaben-befindet-sich-in': dict({
      'hidden': False,
      'label': 'das-bauvorhaben-befindet-sich-in',
      'value': list([
        dict({
          'displayValue': 'gewaesserraum',
          'value': 'das-bauvorhaben-befindet-sich-in-gewaesserraum',
        }),
        dict({
          'displayValue': 'gewaesserschutzbereich',
          'value': 'das-bauvorhaben-befindet-sich-in-gewaesserschutzbereich',
        }),
        dict({
          'displayValue': 'pruefperimeter-bodenaushub',
          'value': 'das-bauvorhaben-befindet-sich-in-pruefperimeter-bodenaushub',
        }),
        dict({
          'displayValue': 'gefahrenzone',
          'value': 'das-bauvorhaben-befindet-sich-in-gefahrenzone',
        }),
        dict({
          'displayValue': 'risikokataster-chemiesicherheit',
          'value': 'das-bauvorhaben-befindet-sich-in-risikokataster-chemiesicherheit',
        }),
      ]),
    }),
  })
# ---
# name: test_ag_client[scenario33]
  dict({
    'das-bauvorhaben-befindet-sich-in': dict({
      'hidden': False,
      'label': 'das-bauvorhaben-befindet-sich-in',
      'value': list([
        dict({
          'displayValue': 'gewaesserraum',
          'value': 'das-bauvorhaben-befindet-sich-in-gewaesserraum',
        }),
        dict({
          'displayValue': 'gewaesserschutzbereich',
          'value': 'das-bauvorhaben-befindet-sich-in-gewaesserschutzbereich',
        }),
        dict({
          'displayValue': 'kataster-belasteter-standorte',
          'value': 'das-bauvorhaben-befindet-sich-in-kataster-belasteter-standorte',
        }),
        dict({
          'displayValue': 'pruefperimeter-bodenaushub',
          'value': 'das-bauvorhaben-befindet-sich-in-pruefperimeter-bodenaushub',
        }),
        dict({
          'displayValue': 'gefahrenzone',
          'value': 'das-bauvorhaben-befindet-sich-in-gefahrenzone',
        }),
        dict({
          'displayValue': 'risikokataster-chemiesicherheit',
          'value': 'das-bauvorhaben-befindet-sich-in-risikokataster-chemiesicherheit',
        }),
      ]),
    }),
  })
# ---
# name: test_ag_client[scenario34]
  dict({
    'das-bauvorhaben-befindet-sich-in': dict({
      'hidden': False,
      'label': 'das-bauvorhaben-befindet-sich-in',
      'value': list([
        dict({
          'displayValue': 'gewaesserraum',
          'value': 'das-bauvorhaben-befindet-sich-in-gewaesserraum',
        }),
        dict({
          'displayValue': 'gewaesserschutzbereich',
          'value': 'das-bauvorhaben-befindet-sich-in-gewaesserschutzbereich',
        }),
        dict({
          'displayValue': 'pruefperimeter-bodenaushub',
          'value': 'das-bauvorhaben-befindet-sich-in-pruefperimeter-bodenaushub',
        }),
        dict({
          'displayValue': 'gefahrenzone',
          'value': 'das-bauvorhaben-befindet-sich-in-gefahrenzone',
        }),
        dict({
          'displayValue': 'risikokataster-chemiesicherheit',
          'value': 'das-bauvorhaben-befindet-sich-in-risikokataster-chemiesicherheit',
        }),
      ]),
    }),
  })
# ---
# name: test_ag_client[scenario35]
  dict({
    'das-bauvorhaben-befindet-sich-in': dict({
      'hidden': False,
      'label': 'das-bauvorhaben-befindet-sich-in',
      'value': list([
        dict({
          'displayValue': 'gewaesserraum',
          'value': 'das-bauvorhaben-befindet-sich-in-gewaesserraum',
        }),
        dict({
          'displayValue': 'gewaesserschutzbereich',
          'value': 'das-bauvorhaben-befindet-sich-in-gewaesserschutzbereich',
        }),
        dict({
          'displayValue': 'pruefperimeter-bodenaushub',
          'value': 'das-bauvorhaben-befindet-sich-in-pruefperimeter-bodenaushub',
        }),
        dict({
          'displayValue': 'gefahrenzone',
          'value': 'das-bauvorhaben-befindet-sich-in-gefahrenzone',
        }),
        dict({
          'displayValue': 'risikokataster-chemiesicherheit',
          'value': 'das-bauvorhaben-befindet-sich-in-risikokataster-chemiesicherheit',
        }),
      ]),
    }),
    'kantonsstrassen': dict({
      'displayValue': 'ja',
      'hidden': False,
      'label': 'kantonsstrassen',
      'value': 'kantonsstrassen-ja',
    }),
  })
# ---
# name: test_ag_client[scenario36]
  dict({
    'waldareal': dict({
      'displayValue': 'ja',
      'hidden': False,
      'label': 'waldareal',
      'value': 'waldareal-ja',
    }),
  })
# ---
# name: test_ag_client[scenario37]
  dict({
    'das-bauvorhaben-befindet-sich-in': dict({
      'hidden': False,
      'label': 'das-bauvorhaben-befindet-sich-in',
      'value': list([
        dict({
          'displayValue': 'gewaesserraum',
          'value': 'das-bauvorhaben-befindet-sich-in-gewaesserraum',
        }),
        dict({
          'displayValue': 'gewaesserschutzbereich',
          'value': 'das-bauvorhaben-befindet-sich-in-gewaesserschutzbereich',
        }),
        dict({
          'displayValue': 'pruefperimeter-bodenaushub',
          'value': 'das-bauvorhaben-befindet-sich-in-pruefperimeter-bodenaushub',
        }),
        dict({
          'displayValue': 'eisenbahnlinie',
          'value': 'das-bauvorhaben-befindet-sich-in-eisenbahnlinie',
        }),
        dict({
          'displayValue': 'gefahrenzone',
          'value': 'das-bauvorhaben-befindet-sich-in-gefahrenzone',
        }),
        dict({
          'displayValue': 'risikokataster-chemiesicherheit',
          'value': 'das-bauvorhaben-befindet-sich-in-risikokataster-chemiesicherheit',
        }),
      ]),
    }),
  })
# ---
# name: test_ag_client[scenario38]
  dict({
    'das-bauvorhaben-befindet-sich-in': dict({
      'hidden': False,
      'label': 'das-bauvorhaben-befindet-sich-in',
      'value': list([
        dict({
          'displayValue': 'gewaesserraum',
          'value': 'das-bauvorhaben-befindet-sich-in-gewaesserraum',
        }),
        dict({
          'displayValue': 'kataster-belasteter-standorte',
          'value': 'das-bauvorhaben-befindet-sich-in-kataster-belasteter-standorte',
        }),
        dict({
          'displayValue': 'gefahrenzone',
          'value': 'das-bauvorhaben-befindet-sich-in-gefahrenzone',
        }),
      ]),
    }),
  })
# ---
# name: test_ag_client[scenario3]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Landschaften und Naturdenkmäler von nationaler Bedeutung (BLN), Landschaftsschutzzone',
    }),
  })
# ---
# name: test_ag_client[scenario4]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Landschaftsschutzzone innerhalb Dekretsgebiet (kant. NP), Landschaften und Naturdenkmäler von nationaler Bedeutung (BLN), Wildtierkorridor (AG-R12), Naturschutzgebiet von kant. Bedeutung (NkB)',
    }),
  })
# ---
# name: test_ag_client[scenario5]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Wasserzone innerhalb Dekretsgebiet (kant. NP), Landschaften und Naturdenkmäler von nationaler Bedeutung (BLN), Wasser- und Zugvogelreservate von internationaler und nationaler Bedeutung, Auengebiet von nationaler Bedeutung',
    }),
  })
# ---
# name: test_ag_client[scenario6]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Landschaften und Naturdenkmäler von nationaler Bedeutung (BLN), Trockenwiese oder -weide von nationaler Bedeutung, Hecke im Kulturlandplan, Naturschutzgebiet von kant. Bedeutung (NkB)',
    }),
  })
# ---
# name: test_ag_client[scenario7]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Waldnaturschutzinventar (Zurlindeninsel), Amphibienlaichgebiet von nationaler Bedeutung, Auengebiet von nationaler Bedeutung, Weitere überlagerte Schutzzone, Feucht- und Nassstandort',
    }),
  })
# ---
# name: test_ag_client[scenario8]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Amphibienlaichgebiet von nationaler Bedeutung',
    }),
  })
# ---
# name: test_ag_client[scenario9]
  dict({
    'ueberlagerte-schutzzonen-und-schutzobjekte': dict({
      'hidden': False,
      'label': None,
      'value': 'Hecke im Kulturlandplan, Landschaftsschutzzone',
    }),
  })
# ---
